src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
-
+function hasIncorrectSize(...sizes: number[]): boolean {
+  return sizes.some((size) => size <= 0);
 }
 
-export class Triangle {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (hasIncorrectSize(this.sideA, this.sideB, this.sideC)) {
+      throw new Error('Triangle\'s side size is incorrect');
+    }
+
+    const longestSide = Math.max(this.sideA, this.sideB, this.sideC);
+    const oppositeSides = (this.sideA + this.sideB + this.sideC) - longestSide;
+
+    if (longestSide >= oppositeSides) {
+      throw new Error('Not a triangle!');
```

Comment:
I think you can make error text more descriptive
![image](https://user-images.githubusercontent.com/96409563/216812383-b8590dda-35c6-479e-88fe-1a02b6db9ded.png)